eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -74,23 +80,53 @@ steps:
       AZURE_VERSION_OVERRIDE_TESTS: ${{ parameters.TestVersionSupport }}
     condition: and(succeeded(), eq(variables['TestVersionSupport'], 'true'))
 
-  # we want to run this when TestFromSource isn't true
-  - task: Maven@3
-    ${{ if ne(parameters.VersionOverride, '') }}:
-      displayName: '${{ parameters.VersionOverride }} - Run tests'
-    ${{ else }}:
-      displayName: 'Run tests'
-    inputs:
-      mavenPomFile: pom.xml
-      options: '$(DefaultTestOptions) -DAZURE_TEST_DEBUG=$(IsDebug) -Dparallel-test-playback=${{ parameters.ParallelTestPlayback }} ${{ parameters.TestOptions }} -T ${{ parameters.TestParallelization }}'
-      mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-      javaHomeOption: 'JDKVersion'
-      jdkVersionOption: $(JavaTestVersion)
-      jdkArchitectureOption: 'x64'
-      publishJUnitResults: false
-      goals: ${{ parameters.TestGoals }}
-    env: ${{ parameters.TestEnvVars }}
-    condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
+  - ${{ if eq('true', parameters.UseFederatedAuth) }}:
+    # Remove preceeding "1." in the version specified by $(JavaTestVersion)
+    # e.g. 1.8 -> 8
+    # This is required to set JAVA_HOME correctly for the Java version
+    - pwsh: |
+        $javaTestVersion = "$(JavaTestVersion)" -replace '^\d*?\.', ''
+        $javaHomeValue = [System.Environment]::GetEnvironmentVariable("JAVA_HOME_$javaTestVersion`_X64")
+        Write-Host "Setting PowerShellJavaHome to '$javaHomeValue'"
+        Write-Host "##vso[task.setvariable variable=PowerShellJavaHome;]$javaHomeValue"
+      displayName: 'Set PowerShellJavaHome'
+
+    - task: AzurePowerShell@5
+      condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
+      ${{ if ne(parameters.VersionOverride, '') }}:
+        displayName: '${{ parameters.VersionOverride }} - Run tests (Federated Auth)'
+      ${{ else }}:
+        displayName: 'Run tests (Federated Auth)'
+      inputs:
+        azureSubscription: ${{ parameters.ServiceConnection }}
+        azurePowerShellVersion: LatestVersion
+        pwsh: true
+        ScriptType: InlineScript
+        Inline: |
+          mvn --% -f $(Build.SourcesDirectory)/pom.xml $(DefaultTestOptions) -DAZURE_TEST_DEBUG=$(IsDebug) -Dparallel-test-playback=${{ parameters.ParallelTestPlayback }} ${{ parameters.TestOptions }} -T ${{ parameters.TestParallelization }} ${{ parameters.TestGoals }}
```

Comment:
Can we build the options once and use it in both places? This feels like a maintenance risk.

Reply:
We'll eventually move away from `AzurePowerShell` so this whole thing can be removed when we do that.